webapp-django/crashstats/crashstats/static/crashstats/js/socorro/pending.js

```diff
@@ -1,47 +1,38 @@
+var Checker = (function() {
+    var intervalTime = 5 * 1000;
+    var checkInterval;
+    var totalTime = 0;
 
-/* Javascript for the Pending Reports page */
-
-// Begin the timer and Ajax calls for reports
-var original_seconds = 30;
-var seconds = original_seconds;
-var number_calls = 1;
-
-// Maintain the time in seconds, and make an ajax call every 30 seconds
-function pendingReportTimer(url){
-    if (seconds == 0){
-        $('#next_attempt').hide();
-        $('#processing').show();
-
-        // Upon the third attempt, state that this failed
-        if (number_calls == 10) {
-            $('#checking').hide();
-            $('#fail').show();
-        } else {
-            pendingReportCheck(url);
-            number_calls += 1;
-            seconds = original_seconds;
-            $('#counter').html(original_seconds);
-            setTimeout("pendingReportTimer(\""+url+"\")",1000);
+    return {
+        startChecking: function(crashID) {
+            checkInterval = setInterval(function() {
+                $.get('/api/ProcessedCrash/', {crash_id: crashID})
+                .then(function() {
+                    clearInterval(checkInterval);
+                    // If it exists, we can reload the page we're one.
+                    $('.pending .searching').hide();
+                    $('.pending .found').fadeIn(500, function() {
```

Comment:
I've read it's good practice to make animations 350ms at most. I'm actually using ``300`` in most places, so I'd recommend you use the same value here.